File: app/src/main/java/com/apt/delegate/source/IOtherDataSource.java
package com.apt.delegate.source;

import com.annotation.Delegate;
import com.annotation.SingleDelegate;

@SingleDelegate(
        classNameImpl = "",
        delegate = @Delegate(
                delegatePackage = "com.apt.delegate.source.delegate",
                delegateClassName = "OtherDelegate",
                delegateSimpleName = "otherDelegate"
        ))
public interface IOtherDataSource {

    Object getOtherData();
}

File: app/src/main/java/com/apt/delegate/source/IOtherDataSource.kt
package com.apt.delegate.source

import com.annotation.Delegate
import com.annotation.SingleDelegate

@SingleDelegate(
        classNameImpl = "",
        delegate = Delegate(
                delegatePackage = "com.apt.delegate.source.delegate",
                delegateClassName = "OtherDelegate",
                delegateSimpleName = "otherDelegate"))
interface IOtherDataSource {

    fun getOtherData(): Any?
}
